app/controllers/posts_controller.rb

```diff
@@ -5,6 +5,15 @@ class PostsController < ApplicationController
   # GET /posts.json
   def index
     @posts = Post.all
+    @posts_done_homework = Post.where(done_homework: true, user_id: current_user.id) 
+    @posts_pending_homework =  Post.where(done_homework: false, user_id: current_user.id) 
```

Comment:
When you make the relations correctly on your migrations and models, all of this kind of code 
will simplify to `@posts_done_homework = Post.where(done_homework: true, user: current_user)` you won't have to explicitly to specify that wich field keeps the relations between your models.